circuits/ts/proofs.ts

```diff
@@ -0,0 +1,109 @@
+import { readFileSync, writeFileSync, unlinkSync, existsSync, mkdirSync, rmdirSync } from "fs";
+import { join } from "path";
+import { execSync } from "child_process";
+import { tmpdir } from "os";
+import { zKey, groth16 } from "snarkjs";
+import { stringifyBigInts } from "maci-crypto";
+import { cleanThreads, isArm } from "./utils";
+
+/**
+ * Generate a zk-SNARK proof
+ * @dev if running on a intel chip we use rapidsnark for
+ * speed - on the other hand if running on ARM we need to use
+ * snark and a WASM witness
+ * @param inputs - the inputs to the circuit
+ * @param zkeyPath - the path to the zkey
+ * @param rapidsnarkExePath - the path to the rapidnsark binary
+ * @param witnessExePath - the path to the compiled witness binary
+ * @param wasmPath - the path to the wasm witness
+ * @param silent - whether we want to print to the console or not
+ * @returns the zk-SNARK proof
+ */
+export const genProof = async (
+  inputs: string[],
+  zkeyPath: string,
+  rapidsnarkExePath?: string,
+  witnessExePath?: string,
+  wasmPath?: string,
+  silent = true,
+): Promise<any> => {
+  // if we are running on an arm chip we can use snarkjs directly
+  if (isArm()) {
+    const { proof, publicSignals } = await groth16.fullProve(inputs, wasmPath, zkeyPath);
+    return { proof, publicSignals };
+  }
+  // intel chip flow (use rapidnsark)
+  // Create tmp directory
+  const tmpPath = join(tmpdir(), `tmp-${Date.now()}`);
+  mkdirSync(tmpPath, { recursive: true });
+
+  const inputJsonPath = join(tmpPath, "input.json");
+  const outputWtnsPath = join(tmpPath, "output.wtns");
+  const proofJsonPath = join(tmpPath, "proof.json");
+  const publicJsonPath = join(tmpPath, "public.json");
+
+  // Write input.json
+  const jsonData = JSON.stringify(stringifyBigInts(inputs));
+  writeFileSync(inputJsonPath, jsonData);
+
+  // Generate the witness
+  const witnessGenCmd = `${witnessExePath} ${inputJsonPath} ${outputWtnsPath}`;
+
+  try {
+    execSync(witnessGenCmd, { stdio: silent ? "ignore" : "pipe" });
+
+    if (!existsSync(outputWtnsPath)) {
+      throw new Error("Error executing " + witnessGenCmd);
+    }
```

Comment:
This error will be catched with catch clause below and the error will be: `"Error executing " + witnessGenCmd + " " + Error executing " + witnessGenCmd`